controllers/application/controller.go

```diff
@@ -91,6 +94,16 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req reconcile.Req
 		return reconcile.Result{}, err
 	}
 
+	err = r.validateApplicationSpec(application)
+	if err != nil {
+		r.GetRecorder().Eventf(
+			application,
+			corev1.EventTypeWarning, "InvalidApplication",
+			"Application was not valid, error: %s", err.Error(),
```

Comment:
```suggestion
			"Application failed validation and was rejected, error: %s", err.Error(),
```
How do you feel about this text?

Reply:
better wording indeed :)